test_common/harness/parseParameters.cpp

```diff
@@ -198,6 +205,32 @@ int parseCustomParam(int argc, const char *argv[], const char *ignore)
                 return -1;
             }
         }
+        else if (!strncmp(argv[i],
+                          "--invalid-object-scenarios=", i_object_length))
+        {
+            if (strlen(argv[i]) > i_object_length)
+            {
+                delArg++;
+                gInvalidObject = 0;
```

Comment:
It looks like the only way to specify no validity checking at all is by specifying anything at all, like `--invalid-object-scenarios=foo`. If we want to provide this capability, we should add an option like "none" and enforce that it's specified correctly.
Otherwise, since "nullptr" checks are always required, it seems like the whole scheme could be simplified greatly by providing a single boolean argument like `--check-object-types` or such.

Reply:
It would be easy to add a "none" option but given that handling the "nullptr" case is mandatory and very simple, I'm not sure I see the point. Do you have a use in mind?

It was designed to take a list so the mechanism is extensible. As part of working group discussions (see https://github.com/KhronosGroup/OpenCL-Docs/issues/482), we have identified several other scenarios for which we may want to add support to the CTS in the future.

Reply:
I may have misinterpreted the intent.  What I was reacting to is the fact that nullptr checks are mandatory, but it still needs to be provided explicitly as an argument. That led me to believe that we wanted to provide the ability to turn off these checks for whatever reason. That can be already be done by omitting the argument altogether, so disregard that part of my comment. 
The question I still have is why we need this argument capability (vs just testing whatever the spec requires for each API) and this level of granularity (vs a binary check/no-check option). Is the idea to provide optional stricter tests for vendors that have implemented those checks?

Reply:
You got it. It's a lot less work to make this generic in the CTS than it is to write separate stricter tests and different vendors will handle different cases.

Reply:
The plan is also to use those stricter options to test future validation layers.